### PR TITLE
Add cass for IMA test environment setup

### PR DESCRIPTION
**Pull request by wnereiz**
#### Add testing for MOK enrollment with mokutil

Add testing for IMA test environment setup. First, generate key pair, sign the kernel, and import the certificate by mokutil. Then reboot to enroll the MOK certificate, and boot with signed kernel.
    
Variable MOK_CERTCONF is involved to specify config file for openssl certificate generating. If not specified, the default value is data/openssl/gencert_conf/mok_cert.conf

#### Setup IMA/EVM test environment

Setup /etc/fstab and grub booting options. Install necessary IMA packages. Reboot to enable the settings.

---
- Related ticket: 
  - https://progress.opensuse.org/issues/45662
  - https://progress.opensuse.org/issues/45701
- Verification run: 
  - SLE15 SP1: http://10.67.17.9/tests/259
  - openSUSE Tumbleweed: http://10.67.17.9/tests/260

### CODE REVIEW

File: tests/security/ima/ima_setup.pm [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    # Add 'iversion' to fstab mount options
+    assert_script_run "awk -i inplace '{if(\$3 == \"ext4\") \$4=\$4\",iversion\"; print}' /etc/fstab";
+
+    # Add 'rootflags=iversion' to grub2 boot menu
+    add_grub_cmdline_settings('rootflags=iversion', 1);
+    assert_script_run "grub2-mkconfig -o /boot/grub2/grub.cfg";
```
Comment (pevik) [inline tests/security/ima/ima_setup.pm]: `assert_script_run "grub2-mkconfig -o /boot/grub2/grub.cfg";` is not needed as it's done by `1` parameter in `add_grub_cmdline_settings()` call.

File: tests/security/ima/ima_setup.pm [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+use strict;
+use testapi;
+use utils;
+use bootloader_setup 'add_grub_cmdline_settings';
+use power_action_utils "power_action";
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
```
Comment (pevik) [inline tests/security/ima/ima_setup.pm]: I guess you prefer not using virtio console (`select_serial_terminal()`) :).
BTW I did e3fc5d807 ("security: Migrate SELinux tests to select_serial_terminal()") commit in hope that you like it and I/you would migrate rest of the security tests to virtio console. It's up to you, if you don't like it you can revert this change (probably make sense to be consistent on all security tests).

File: tests/security/ima/ima_setup.pm (hunk 1 above)
Comment (wnereiz) [reply tests/security/ima/ima_setup.pm]: Thanks for the tip, modified

File: tests/security/mokutil_sign.pm [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+use registration 'add_suseconnect_product';
+use power_action_utils "power_action";
+
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    if (is_sle('>12')) {
```
Comment (okurz) [inline tests/security/mokutil_sign.pm]: that would trigger for each service pack of 12 as well I assume. I guess it should be '>=15'

File: tests/security/mokutil_sign.pm [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    my $cert_pemdata = get_var("MOK_CERTCONF") // "openssl/gencert_conf/mok_cert.conf";
+    assert_script_run "wget --quiet " . data_url($cert_pemdata) . " -O $cert_cfg";
+
+    # Use default expiration days (1 month)
+    assert_script_run("openssl req -new -x509 -newkey rsa -keyout $pri_key -out $cert_pem -nodes -config $cert_cfg");
+    assert_script_run("openssl pkcs12 -export -inkey $pri_key -in $cert_pem -name kernel_cert -out $cert_p12 -passout pass:$key_pw");
+    assert_script_run("openssl x509 -in $cert_pem -outform der -out $cert_der");
+
+    # certutil -N does not support password in command arguement, so the
```
Comment (okurz) [inline tests/security/mokutil_sign.pm]: s/arguement/argument/

File: tests/security/ima/ima_setup.pm (hunk 2 above)
Comment (wnereiz) [reply tests/security/ima/ima_setup.pm]: I would like to try. I have switched `ima_setup.pm` to serial_terminal. But I can not do that for `mokutil_sign.pm`, it always failed when handling interactive commands with script_run_interactive (which I just created and submitted to lib/utils.pm), see http://10.67.17.9/tests/252#step/mokutil_sign/29

It looks like the serial console always waiting for **'#'** which is the prompt by default, but in the interactive command it is not usually use it.

Currently, I do not have enough time for more investigation. So I would like to keep this test case there. Do you have any ideas?

File: tests/security/mokutil_sign.pm [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            },
+        ],
+        20
+    );
+    assert_script_run("pk12util -d $work_dir -i $cert_p12 -K $cdb_pw -W $key_pw");
+    assert_script_run("ls $work_dir | tee /dev/$serialdev");
+
+    my $kern = script_output("ls /boot/vmlinuz-*-default");
+
+    # Remove existed signature and sign with new one to ensure it will boot
```
Comment (okurz) [inline tests/security/mokutil_sign.pm]: s/existed/existing/

Comment (wnereiz) [reply tests/security/mokutil_sign.pm]: Sorry my typo, modified

File: tests/security/mokutil_sign.pm (hunk 3 above)
Comment (wnereiz) [reply tests/security/mokutil_sign.pm]: My mistake, modified!
